buildpacks/metrics-agent/src/layers/install_agentmon.rs

```diff
@@ -0,0 +1,329 @@
+        .build()
+    }
+
+    fn existing_layer_strategy(
+        &self,
+        _context: &libcnb::build::BuildContext<Self::Buildpack>,
+        layer_data: &libcnb::layer::LayerData<Self::Metadata>,
+    ) -> Result<libcnb::layer::ExistingLayerStrategy, <Self::Buildpack as libcnb::Buildpack>::Error>
+    {
+        if let Some(old_url) = &layer_data.content_metadata.metadata.download_url {
+            let url = get_latest_url()
+                .map_err(DownloadAgentmonError::CannotGetLatestUrl)
+                .map_err(MetricsAgentError::DownloadAgentmonError)?;
+
+            if old_url == &url {
+                self.section.say("Using cache");
+                Ok(ExistingLayerStrategy::Keep)
+            } else {
+                let url = build_output::fmt::value(url);
+                self.section
+                    .say_with_details("Clearing cache", format!("Updated url {url}"));
+                Ok(ExistingLayerStrategy::Recreate)
+            }
+        } else {
+            self.section
+                .say_with_details("Clearing cache", "No url found in metadata");
+
+            Ok(ExistingLayerStrategy::Recreate)
+        }
+    }
+
+    fn migrate_incompatible_metadata(
+        &self,
+        _context: &libcnb::build::BuildContext<Self::Buildpack>,
+        _metadata: &GenericMetadata,
+    ) -> Result<
+        libcnb::layer::MetadataMigration<Self::Metadata>,
+        <Self::Buildpack as libcnb::Buildpack>::Error,
+    > {
+        self.section
+            .say_with_details("Clearing cache", "invalid metadata");
+
+        Ok(libcnb::layer::MetadataMigration::RecreateLayer)
+    }
+}
+
+fn write_execd(agentmon_path: &Path, layer_path: &Path) -> Result<PathBuf, DownloadAgentmonError> {
+    let agentmon_path = agentmon_path
+        .canonicalize()
+        .map_err(DownloadAgentmonError::CouldNotOpenFile)?;
+    let agentmon_path = agentmon_path.display();
+
+    // This script boots and runs agentmon in a loop
+    let agentmon_loop = {
+        let script = layer_path.join("agentmon_loop");
+
+        // Copy compiled binary from `bin/agentmon_loop.rs` to layer
+        fs_err::copy(additional_buildpack_binary_path!("agentmon_loop"), &script)
+            .map_err(DownloadAgentmonError::CouldNotWriteDestinationFile)?;
+
+        script
+            .canonicalize()
+            .map_err(DownloadAgentmonError::CouldNotOpenFile)?
+    };
+
+    // We use the exec.d to boot a process. This script MUST exit though as otherwise
+    // The container would never boot. To handle this we intentionally leak a process
+    let execd_script = {
+        let execd_script = layer_path.join("agentmon_exec.d");
+        let log_file = layer_path.join("background.log");
+        fs_err::write(&log_file, "")
+            .map_err(DownloadAgentmonError::CouldNotCreateDestinationFile)?;
+
+        let log_file = log_file.display();
+        let agentmon_loop = agentmon_loop.display();
```

Comment:
Nit: this made me pause for a moment - "how can the loop be displayed? And why it is re-assigned?" Renaming would go a long way here.